src/back/glsl.rs

```diff
@@ -759,34 +960,42 @@ fn write_constant(
                 TypeInner::Matrix { columns, rows, .. } =>
                     format!("mat{}x{}", columns as u8, rows as u8,),
                 TypeInner::Struct { .. } => builder.structs.get(&constant.ty).unwrap().clone(),
-                TypeInner::Array { .. } => write_type(constant.ty, &module.types, builder.structs)?,
+                TypeInner::Array { .. } =>
+                    write_type(constant.ty, &module.types, builder.structs, None, features)?,
                 _ =>
                     return Err(Error::Custom(format!(
                         "Cannot build constant of type {}",
-                        write_type(constant.ty, &module.types, builder.structs)?
+                        write_type(constant.ty, &module.types, builder.structs, None, features)?
                     ))),
             },
             components
                 .iter()
-                .map(|component| write_constant(&module.constants[*component], module, builder))
+                .map(|component| write_constant(
+                    &module.constants[*component],
+                    module,
+                    builder,
+                    features
+                ))
                 .collect::<Result<Vec<_>, _>>()?
                 .join(","),
         ),
     })
 }
 
-fn write_type<'a>(
+fn write_type(
     ty: Handle<Type>,
-    types: &'a Arena<Type>,
-    structs: &'a FastHashMap<Handle<Type>, String>,
+    types: &Arena<Type>,
+    structs: &FastHashMap<Handle<Type>, String>,
+    block: Option<String>,
+    features: SupportedFeatures,
 ) -> Result<String, Error> {
```

Comment:
Let's return `Cow<'static, str>` instead of `String` here, it would allow us to avoid heap allocating constant strings